src/components/form/utils/submitFormData.ts

```diff
@@ -0,0 +1,25 @@
+import { checkAllForm } from "./formValidation";
+
+export const submitFormData = (event: Event) => {
+  const form = event.target as HTMLFormElement;
+  if (form != null) {
+    if (checkAllForm(form)) {
+      printFormData(form);
+    }
+  }
+};
+
+const printFormData = (form: HTMLFormElement) => {
+  const inputs = form.querySelectorAll("input");
+
+  if (inputs != null) {
+    const data: Record<string, string> = {};
+
+    inputs.forEach((input) => {
+      const { name } = input;
+      data[name] = input.value;
+    });
+
+    console.log(data);
```

Comment:
Здесь нам скорее длина массива важна. Если элементов нет подходящих, то вернётся пустой массив (который на `null` проверку пройдёт), а уже когда запустится forEach выяснится, что массив пустой. 

Как вариант можно сделать проверку на обратное условие и в блоке if сделать return, а если проверку не пройдёт, то текущий вариант кода, только вместо forEach можно сразу использовать reduce:

```suggestion
  if (!inputs || inputs.length === 0) {
     return
  }
  const data: Record<string, string> = inputs.reduce((total, input) => {
    const { name } = input;
    total[name] = input.value;
    return total;
  }, {});

  console.log(data);
```